ludwig/schema/features/augmentation/utils.py

```
@@ -0,0 +1,145 @@
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to an augmentation config.
+    """
+
+    class AugmentationContainerMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a list for a valid augmentation config from the
+        augmentation_registry and creates a corresponding JSON schema for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            assert isinstance(value, list), "Augmentation config must be a list."
+
+            augmentation_list = []
+            for augmentation in value:
+                augmentation_op = augmentation[TYPE]
+                augmentation_cls = get_augmentation_cls(augmentation_op)
+                pre = augmentation_cls()
+                try:
+                    augmentation_list.append(pre.Schema().load(augmentation))
+                except (TypeError, ValidationError) as error:
+                    raise ValidationError(
+                        f"Invalid augmentation params: {value}, see `{pre}` definition. Error: {error}"
+                    )
+            return augmentation_list
+
+        @staticmethod
+        def _jsonschema_type_mapping():
+            return get_augmentation_jsonschema()
+
+    try:
+        if default:
+            augmentation_list = []
+            for augmentation in default:
+                augmentation_op = augmentation[TYPE]
+                augmentation_cls = get_augmentation_cls(augmentation_op)
+                pre = augmentation_cls()
+                try:
+                    augmentation_list.append(pre.Schema().load(augmentation))
+                except (TypeError, ValidationError) as error:
+                    raise ValidationError(
+                        f"Invalid augmentation params: {default}, see `{pre}` definition. Error: {error}"
+                    )
+            load_default = dump_default = augmentation_list
+        else:
```

if `default` is `None` we should just raise an error rather than doing any kind of fallback logic.

---

@ksbrar   I think the change for this comment https://github.com/ludwig-ai/ludwig/pull/2925/files#r1070381132  may have addressed this.

What do you think?

---

@ksbrar    Nevermind...I misunderstood your commet.  I re-read it and see where I went astray.  I'll make the suggested fix.